### PR TITLE
2128/confirm prompt validator not invoked as expected

### PR DESCRIPTION
**Pull request by longieirl**
Fix for https://github.com/SAP/open-ux-tools/issues/2128

- The confirm SCP prompt has a validator method attached which is not invoked for all flows i.e. CLI and YUI, refer to [inquirer API](https://github.com/SBoudrias/Inquirer.js/tree/main/packages/confirm). However, the abap-inq was written as if this `validator` was invoked for both YUI and CLI. This PR removes the validator method and ensures the `scp` prompt state is maintained correctly by adding a scp setter prompt to handle both flows, CLI and YUI.
- If the user selects a SAP system, the client prompt will no longer be shown, this causes issues with the connectivity flow if the user tries to switch client ID's as then the saved system properties will not be used
- Tests added
- Validated locally on VSC and on BAS

Example1;

ui5.yaml
```yaml
        backend:
          - scp: true
            path: /sap
            url: https://example.ondemand.com
```
Prompts;
```bash
Adding deploy-config to the project.
info Add:Deploy-config Using: @sap/fiori:deploy-config
? Select Target system abap-05 (BTP) [tester@example]
? SAPUI5 ABAP Repository (ZTEST) 
```

Example2;

ui5.yaml
```yaml
        backend:
          - scp: false
            path: /sap
            url: https://example.ondemand.com
            client: 300
```
Prompts;
```bash
Adding deploy-config to the project.
info Add:Deploy-config Using: @sap/fiori:deploy-config
? Select Target system abap-05 (BTP) [john.long@sap.com]
? SAPUI5 ABAP Repository (ZTEST) 
```

Example3;

Prompts;
```bash
info Add:Deploy-config Using: @sap/fiori:deploy-config
? Select Target system Enter Target System URL
? Target System URL https://example.net.sap:44300
? Is this an SAP Business Technology Platform system? Yes
? SAPUI5 ABAP Repository (ZTEST) 
```

Example4;
```bash
info Add:Deploy-config Using: @sap/fiori:deploy-config
? Select Target system Enter Target System URL
? Target System URL https://example.net.sap:44300
? Is this an SAP Business Technology Platform system? No
? Client (Use arrow keys)
❯ Use project defined client 300 
  Enter client 
  Use default system client 
```

Example5;
```bash
info Add:Deploy-config Using: @sap/fiori:deploy-config
? Select Target system Enter Target System URL
? Target System URL https://example.devint.net.sap:44300
? Is this an SAP Business Technology Platform system? No
? Enter client 
```

Screenshot 2024-09-17 at 16 32 20

Screenshot 2024-09-17 at 16 32 06

### CODE REVIEW

**Review by cianmSAP — APPROVED** [review]
Update looks good
Not manually tested
Tests to cover
Changeset ✅

File: packages/abap-deploy-config-inquirer/src/prompts/conditions.ts [hunk 1]
```diff
@@ -53,52 +52,45 @@ export function showScpQuestion(previousAnswers: AbapDeployConfigAnswersInternal
...
 /**
- * Client condition to determine if the client question should be shown.
+ * Client condition to determine if the client question should be shown. Validates the SCP confirmation and project configuration properties.
  *
- * @param isS4HanaCloudSystem - is S/4 HANA Cloud system
+ * @param scp - is SCP system
  * @returns boolean
  */
-function showClientCondition(isS4HanaCloudSystem?: boolean): boolean {
+function showClientCondition(scp?: boolean): boolean {
```
Comment (IainSAP) [inline packages/abap-deploy-config-inquirer/src/prompts/conditions.ts:60]: 👍

File: packages/abap-deploy-config-inquirer/src/prompts/questions/abap-target.ts [hunk 2]
```diff
@@ -158,18 +158,36 @@ function getUrlPrompt(
...
+            },
+            default: (): boolean | undefined => backendTarget?.abapTarget?.scp
+        }
+    ];
+    prompts.push({
+        when: (answers: AbapDeployConfigAnswersInternal): boolean => {
+            const scpChoice = answers[abapDeployConfigInternalPromptNames.scp];
+            const targetChoice = answers[abapDeployConfigInternalPromptNames.targetSystem];
+            // scpChoice by default is true so only update state if target system is a URL
+            if (scpChoice && targetChoice === TargetSystemType.Url) {
```
Comment (IainSAP) [inline packages/abap-deploy-config-inquirer/src/prompts/questions/abap-target.ts:179]: Do you need this condition, my understanding is that the BTP (SCP) confirm prompt is only ever asked when the target system type is url so its a redundant check?

Comment (longieirl) [reply packages/abap-deploy-config-inquirer/src/prompts/questions/abap-target.ts:179]: Yes @IainSAP as stated in the PR description, this is to cover both CLI and YUI flows. 

Yes, the SCP prompt is only shown when the target URL option is selected. 

However, based on YUI testing, the state wasn't being updated to reflect the user toggling between Yes | No not did it handle when the user changed their choice from a target URL to a target system. This additional prompt ensures all these flows are covered.

File: packages/abap-deploy-config-inquirer/src/prompts/conditions.ts [hunk 3]
```diff
@@ -53,52 +52,45 @@ export function showScpQuestion(previousAnswers: AbapDeployConfigAnswersInternal
...
     }
-
-    return showClientCondition(isS4HanaCloudSystem);
+    return showClientCondition(previousAnswers?.scp) && previousAnswers?.targetSystem === TargetSystemType.Url;
 }
 
 /**
- * Determines if the client question should be shown.
- * Note: In some instances, when a yaml conf is parsed, double quoted properties i.e. client: "100" are saved as a number instead of a string.
+ * Determines if the client question should be shown under very specific conditions.
```
Comment (IainSAP) [inline packages/abap-deploy-config-inquirer/src/prompts/conditions.ts:81]: Which are the conditions as the code is hard to follow here?

**Review by IainSAP — APPROVED** [review]
Thanks @longieirl 

- Some concern re. understandability of the client prompt condition but defer to the author
- Tests updated
